controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -1222,12 +1209,17 @@ func (c *fakeClient) Create(ctx context.Context, obj runtime.Object, opts ...cli
 }
 
 func createClusterWithControlPlane() (*clusterv1.Cluster, *controlplanev1.KubeadmControlPlane, *unstructured.Unstructured) {
-	cluster := newCluster(&types.NamespacedName{Name: "foo", Namespace: "test"})
+	kcpName := fmt.Sprintf("kcp-foo-%s", util.RandomString(6))
+
+	namespace := "test"
+	cluster := newCluster(&types.NamespacedName{Name: kcpName, Namespace: namespace})
+	cluster.UID = "1"
```

Comment:
The UID field is governed by the apiserver and this is going to be replaced with a random value on creation